src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SNI/SNINpHandle.cs

```diff
@@ -355,8 +355,7 @@ public override uint EnableSsl(uint options)
                 _validateCert = (options & TdsEnums.SNI_SSL_VALIDATE_CERTIFICATE) != 0;
                 try
                 {
-
-                    _sslStream.AuthenticateAsClient(_targetServer);
+                    _sslStream.AuthenticateAsClient(_targetServer, null, SupportedProtocols, true);
```

Comment:
Should this test code also be updated? https://github.com/dotnet/SqlClient/blob/2c2f100d8ab9a1f6ee8c048a260d77e6a738e1e7/src/Microsoft.Data.SqlClient/tests/tools/TDS/TDS.EndPoint/TDSParser.cs#L94

Reply:
Server protocol is set to Tls 1.2; So, the client has to obey it. I don't think modifying it will affect anything.